src/mam4xx/mo_photo.hpp

```diff
@@ -0,0 +1,552 @@
+
+       4) Indices are supplied to select a RSF that is consistent with
+          the reference atmosphere in TUV (for direct comparision of J's).
+          This approach will be replaced in the global model. Here colo3, zenith
+          angle, and altitude will be inputed and the correct entry in the table
+          will be derived.
+  ==============================================================================*/
+
+  // @param[in] sza_in             ! solar zenith angle [degrees]
+  // @param[in] alb_in(nlev)       ! albedo
+  // @param[in]  p_in(nlev)         ! midpoint pressure [hPa]
+  // @param[in]  t_in(nlev)         ! Temperature profile [K]
+  // @param[in]  colo3_in(nlev)     ! o3 column density [molecules/cm^3]
+  // @param[in]  xsqy
+  // @param[in]  sza
+  // @param[in]  del_sza
+  // @param[in]  alb
+  // @param[in]  press
+  // @param[in]  del_p
+  // @param[in]  colo3
+  // @param[in]  o3rat
+  // @param[in]  del_alb
+  // @param[in]  del_o3rat
+  // @param[in]  etfphot
+  // @param[in]  rsf_tab
+  // @param[in]  prs
+  // @param[in]  dprs
+  // @param[in]  nw             wavelengths >200nm
+  // @param[in]  nump           number of altitudes in rsf
+  // @param[in]  numsza         number of zen angles in rsf
+  // @param[in]  numcolo3       number of o3 columns in rsf
+  // @param[in]  numalb         number of albedos in rsf
+  // @param[in]  np_xs          number of pressure levels in xsection table
+  // @param[in]  numj           number of photorates in xsqy, rsf
+  // @param[out]  j_long(:,:)   photo rates [1/s]
+
+  /*----------------------------------------------------------------------
+    ... interpolate table rsf to model variables
+----------------------------------------------------------------------*/
+  const Real nlev = pver;
+  const Real zero = 0;
+  interpolate_rsf(alb_in, sza_in, p_in, colo3_in, nlev, sza, del_sza, alb,
+                  press, del_p, colo3, o3rat, del_alb, del_o3rat, etfphot,
+                  rsf_tab, //  in
+                  nw, nump, numsza, numcolo3, numalb, rsf, psum_l,
+                  psum_u); // out
+
+  /*------------------------------------------------------------------------------
+  ... calculate total Jlong for wavelengths >200nm
+  ------------------------------------------------------------------------------
+  ------------------------------------------------------------------------------
+   LLNL LUT approach to finding temperature index...
+   Calculate the temperature index into the cross section
+   data which lists coss sections for temperatures from
+   150 to 350 degrees K.  Make sure the index is a value
+   between 1 and 201.
+  ------------------------------------------------------------------------------*/
+
+  for (int kk = 0; kk < nlev; kk++) {
+    /*----------------------------------------------------------------------
+      ... get index into xsqy
+     ----------------------------------------------------------------------*/
+    // BAD CONSTANT
```

Comment:
Is the BAD CONSTANT below 148.5?

Reply:
Yes, It does. But, I will add another BAD CONSTANT for 201.